IBX-5948: Product page URLs (#205)
ezsystems/user-documentation: docs/pim/work_with_product_page_urls.md

---
description: You can create manage the system URLs for product items. 
---

# Product page URLs

Every [Product](products.md) has its own URLs that is automatically generated. It is based on a pattern that says which field or fields are used to build the URL (URL alias name pattern).

> Product can have multiple URLs if it is included in many catalogs and exposed to many sites.

Author: As discussed with @kisztof - sentence deleted.

## Product Attributes identifiers

Products have their own [attributes](products.md#attributes) that provide different information about them. 

The following attribute types are available:

- Checkbox
- Color
- Integer
- Measurement
- Selection

You can use attributes identifiers in the URL alias pattern field. 
A customized URL is short and easy to remember. 

It can help to understand what the product is about just by reading the link. 

> It improves the user experience and reduces bounce rates on the website.

Reviewer: I think this part can be moved at the top of the page

---
description: You can create URL alias name pattern and custom URLs for the Products.
---

# Product page URLs

> Every [Product item](products.md) has a system URL that is automatically generated. It is based on a pattern that combines fields and attributes to build the URL (URL alias name pattern).

Reviewer 2: ```suggestion
Every [product item](products.md) has a system URL that is automatically generated. 
It is based on a pattern that combines fields and attributes to build the URL (URL alias name pattern).
```

> When you are creating new [Product Type](create_product_types.md) you can set up Product URL alias name pattern.

Reviewer 2: ```suggestion
When you are creating new [Product Type](create_product_types.md) you can set up product URL alias name pattern.
```

> To do it, fill in all necessary information and choose the attributes that you want to use in URL alias name pattern field and click **Save** button.

Reviewer 2: ```suggestion
To do it, fill in all the necessary information and choose the attributes that you want to use in URL alias name pattern field and click **Save** button.
```

![URL attributes](url_attributes.png)

Now, you can see new URL alias name pattern in the Product Type's view.

![URL alias name pattern.png](url_alias_name_pattern.png)

## Product Attributes identifiers

Products have their own [attributes](products.md#attributes) to define product specification.

The following attribute types can be used in URL alias name pattern field:

- Checkbox
- Color
- Integer
- Selection
- Measurement (single)
- Measurement (range)

| Type | Returned value | Example|
|---|---|---|
| Checkbox | Label of the checkbox is checked | wireless |
| Color | Hex code value | 8b0000 |
| Integer | Integer | 26 |
| Selection | Selected value with configured separator | children-infants |
| Measurement (single) | Single value and unit | 256gb |
| Measurement (range) | Value unit and minimum/maximum values | 20-25cm|

You can use many attributes identifiers as well as fields identifiers in the URL alias pattern field. 

Having keyword-rich URLs also improves the product's visibility in search engine results and boosts the page's ranking in search results.

## Create custom URL

> You can create custom URL for each Product.

Reviewer 2: ```suggestion
You can create custom URL for each product.
```

> Customized URLs are easy to remember, help for SEO optimatization and reduce bounce rates on the website.

Reviewer 2: ```suggestion
Customized URLs are easy to remember, help with SEO optimization and reduce bounce rates on the website.
```

It improves the user experience - you can understand what the product is about just by reading the link. 

To do it, follow the steps:

1\. In the left panel, go to **Product catalog** -> **Products**.

> 2\. Choose the Product from the list.

Reviewer 2: ```suggestion
2\. Choose a product from the list.
```

3\. Click on **URL** tab in the product's view. Here you can see both custom URL aliases and system URL.

![URL tab](url_tab.png)

4\. Click **Add new** button.

5\. Provide custom URL in "URL" field and set up all necessary settings.

![Create custom URL](create_custom_url.png)

6\. Click **Create**.

> Now, in the **URL** tab in the product's view, you can see new custom URL for the Product.

Reviewer 2: ```suggestion
Now, in the **URL** tab in the product's view, you can see new custom URL for the product.
```

---
description: You can create URL alias name pattern and custom URLs for the Products.
---

# Product page URLs

Every [Product item](products.md) has a system URL that is automatically generated. It is based on a pattern that combines fields and attributes to build the URL (URL alias name pattern).

When you are creating new [Product Type](create_product_types.md) you can set up Product URL alias name pattern.

To do it, fill in all necessary information and choose the attributes that you want to use in URL alias name pattern field and click **Save** button.

![URL attributes](url_attributes.png)

Now, you can see new URL alias name pattern in the Product Type's view.

![URL alias name pattern.png](url_alias_name_pattern.png)

## Product Attributes identifiers

> Products have their own [attributes](products.md#attributes) that provide different information about them. 

Reviewer: ```suggestion
Products have their own [attributes](products.md#attributes) to define product specifications. 
```

The following attribute types can be used in URL alias name pattern field:

- Checkbox
- Color
- Integer
- Selection
- Measurement (single)
- Measurement (range)

> !!! note

Reviewer: Not sure we want to keep this :)

    Float atribute is not available while creating URL alias name pattern.

> | Type | Returned value | Example|
> |---|---|---|
> | Checkbox | Label of the checkbox is checked | wireless |
> | Color | Hex code value | 8b0000 |
> | Integer | Integer | 26 |
> | Measurement (single) | Single value and unit | 256gb |
> | Measurement (range) | Value unit and minimum/maximum values | 20-25cm|
> | Selection | Selected value with configured separator | children-infants |

Reviewer: ```suggestion
| Type | Returned value | Example|
|---|---|---|
| Checkbox | Label of the checkbox is checked | wireless |
| Color | Hex code value | 8b0000 |
| Integer | Integer | 26 |
| Selection | Selected value with configured separator | children-infants |
| Measurement (single) | Single value and unit | 256gb |
| Measurement (range) | Value unit and minimum/maximum values | 20-25cm|
```
Changed the order so it's consistent with the list above

You can use many attributes identifiers as well as fields identifiers in the URL alias pattern field. 

> Having keyword-rich URLs also improves the product's visibility in search engine results and boost the page's ranking in search results.

Reviewer: ```suggestion
Having keyword-rich URLs also improves the product's visibility in search engine results and boosts the page's ranking in search results.
```
